src/simulation.hpp

```diff
@@ -2950,7 +2948,46 @@ template <typename problem_t> void AMRSimulation<problem_t>::doDiagnostics()
 
 	for (const auto &diag : m_diagnostics) {
 		if (diag->doDiag(tNew_[0], istep[0])) {
-			diag->processDiag(istep[0], tNew_[0], GetVecOfConstPtrs(diagMFVec), m_diagVars, simulationMetadata_);
+			// Check if this is a DiagPlotfile - if so, call its special writePlotfile method
```

Comment:
I assume it's not currently possible to do this inside processDiag(). Is that because of the face-centered vars, the particles, or both?

Reply:
I think that would be a better long-term solution. What additional parameters are needed?